exercises/transpose/transpose.exs

```diff
@@ -0,0 +1,4 @@
+defmodule Transpose do
+  def transpose(input) do
```

Comment:
Can you add a typespec here? Those are often really helpful for folks as they implement their solutions. At least they know what they are getting, and what they need to return.

Also, a little documentation is always helpful, even if it's duplicative of the information in the README. Many times folks don't even look at that and jump right into the exercise, or if they're viewing the submission on the website, having the documentation as a reminder right there is really helpful.